test/repository_test.rb

```diff
@@ -35,59 +35,38 @@
 
       describe '.persist' do
         describe 'when passed a non-persisted entity' do
-          let(:user) { User.new(name: 'S', age: '3') }
+          let(:unpersisted_user) { User.new(name: 'Don', age: '25') }
 
           it 'should return that entity' do
-            persisted_user = UserRepository.persist(user)
+            persisted_user = UserRepository.persist(unpersisted_user)
 
             persisted_user.id.wont_be_nil
-            persisted_user.name.must_equal(user.name)
-            persisted_user.age.must_equal(user.age.to_i)
+            persisted_user.name.must_equal(unpersisted_user.name)
+            persisted_user.age.must_equal(unpersisted_user.age.to_i)
           end
 
           it 'should coerce attributes' do
-            persisted_user = UserRepository.persist(user)
-            persisted_user.age.must_equal(3)
-          end
-        end
-
-        describe 'when passed a persisted entity' do
-          let(:user)                   { UserRepository.create(User.new(name: 'S')) }
-          let(:already_persisted_user) { UserRepository.persist(user) }
-
-          it 'should return that entity' do
-            UserRepository.persist(already_persisted_user).must_equal(user)
-          end
-        end
-
-        describe 'when non persisted' do
-          before do
-            @user = UserRepository.persist(user)
-          end
-
-          let(:user) { User.new(name: 'S', age: '3') }
-
-          it 'is created' do
-            id = UserRepository.last.id
-            UserRepository.find(id).must_equal(@user)
+            persisted_user = UserRepository.persist(unpersisted_user)
+            persisted_user.age.must_equal(25)
           end
 
-          it 'should coerce attributes' do
-            persisted_user = UserRepository.persist(user)
-            persisted_user.age.must_equal(3)
+          it 'assigns and persist created_at attribute' do
+            persisted_user = UserRepository.persist(unpersisted_user)
+            persisted_user.created_at.must_equal persisted_user.created_at
           end
         end
 
-        describe 'when already persisted' do
-          before do
-            @user1 = UserRepository.create(user1)
+        describe 'when passed a persisted entity' do
+          let(:user)           { UserRepository.create(User.new(name: 'Don', age: '25')) }
```

Comment:
I will edit `User.new(name: 'Don', age: '25')` into `User.new(name: 'Don')`
But line 60 and 61 should be consolidated
I don't think so. We need to keep it to test for this case ` it 'should return that entity' do `